e2e/specs/add-appointments.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { expect } from '@playwright/test';
+import { generateRandomPatient, deletePatient, type Patient } from '../commands';
+import { test } from '../core';
+import { AppointmentsPage } from '../pages';
+
+let patient: Patient;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+});
+
+test('Add appointment for a patient, edit the added appointment and cancel it', async ({ page, api }) => {
+  const appointmentsPage = new AppointmentsPage(page);
+  await test.step('When I click on appointment tab', async () => {
+    await appointmentsPage.goto(patient.uuid);
+  });
+  //test to add appointment
+  await test.step('And I click on the “Add” button', async () => {
+    await page.click('button:has-text("Add")');
+  });
+
+  await test.step('When I select Mobile Clinic location', async () => {
+    await page.locator('#location').selectOption('8d9045ad-50f0-45b8-93c8-3ed4bce19dbf');
+  });
+
+  await test.step('And I select “Outpatient Department” service', async () => {
+    await page.locator('#service').selectOption('Outpatient Department');
+  });
+
+  await test.step('And I make appointment as “Scheduled”', async () => {
+    await page.locator('#appointmentType').selectOption('Scheduled');
+  });
+
+  await test.step('And I set date for tomorrow', async () => {
+    await page.locator('#datePickerInput').fill('18/02/2024');
+  });
+
+  await test.step('And I set the “Duration”', async () => {
+    await page.locator('#duration').fill('60');
+  });
+
+  await test.step('And I add a note', async () => {
+    await page.getByPlaceholder(/Write any additional points here/i).fill('Testing Appointments notes');
+  });
+
+  await test.step('And I click Save button', async () => {
+    await page.getByRole('button', { name: /save and close/i }).click();
+  });
+
+  await test.step('Then I should see a success message', async () => {
+    await expect(page.getByText(/Appointment scheduled/i)).toBeVisible();
+  });
+  // test for editing the appointment
+  await test.step('And I choose the "Edit" option from the popup menu', async () => {
```

Comment:
Break this down to two steps:
- When I click the options kebab menu in the appointment
- And I choose the "Edit" option
This applies to the deletion too.

Also add the following step
- Then I should see the Edit an appointment workspace